s3/replication/common/src/s3replicationcommon/s3_range_read.py

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import aiohttp
+import sys
+from s3replicationcommon.aws_v4_signer import AWSV4Signer
+from s3replicationcommon.log import fmt_reqid_log
+from s3replicationcommon.s3_common import S3RequestState
+from s3replicationcommon.timer import Timer
+
+
+class S3AsyncGetRangeRead:
+    def __init__(self, session, request_id,
+                 bucket_name, object_name,
+                 object_range):
+        """Initialise."""
+        self._session = session
+        # Request id for better logging.
+        self._request_id = request_id
+        self._logger = session.logger
+
+        self._bucket_name = bucket_name
+        self._object_name = object_name
+
+        self._object_range = object_range
+
+        self._remote_down = False
+        self._http_status = None
+
+        self._timer = Timer()
+        self._state = S3RequestState.INITIALISED
+
+    def get_state(self):
+        """Returns current request state."""
+        return self._state
+
+    def get_execution_time(self):
+        """Return total time for GET Object operation."""
+        return self._timer.elapsed_time_ms()
+
+    def get_etag(self):
```

Comment:
those are same as GetObject

Reply:
yes. removed s3_range_read class as I have combined the range read function in GetObject.